modules/common/src/main/scala/surge/kafka/KafkaPartitionShardRouterActor.scala

```diff
@@ -165,39 +173,41 @@ class KafkaPartitionShardRouterActor(
     case GetPartitionRegionAssignments => sender() ! state.partitionRegions
     case GetHealth =>
       sender() ! HealthCheck(name = "shard-router-actor", id = s"router-actor-$hashCode", status = HealthCheckStatus.UP)
-    case msg if extractEntityId.isDefinedAt(msg) => becomeActiveAndDeliverMessage(state, msg)
+    case msg if !canExtractEntityId(msg) =>
```

Comment:
There's a  (pre-existing) unit test that wants to make sure that when the entity id cannot be extracted (because the function provided throws an exception..), the message goes to dead letters.

I don't necessarily agree with this approach since if the extractEntityId provided by the user is flawed, then it's more like a configuration error and the app shouldn't continue running ... plus it's also overly defensive programming ..

Reply:
I'm good with deleting that test and simplifying it the way you had it earlier. I think that test was going more for trying to get coverage than actually testing something valuable and I 100% agree with you

Reply:
I agree, do you mind if we do this cleanup in a subsequent PR and not in this one ?